### ISSUE TITLE
Tables without captions?

### ISSUE THREAD

**Issue by larseggert** [cabo/kramdown-rfc2629#82]
Is it possible to have a table without a caption? If yes, how?

**Comment by cabo** [comment]
In v2, you could give it neither an anchor (#xxx) nor a title.

I.e., just the unadorned table:

```
| Lars' Table | with | some | content |
|           1 |    2 |    3 |       4 |
|           5 |    6 |    7 |       8 |
```

It seems to me this was lost in v3.

**Comment by cabo** [comment]
@levkowetz -- is this still possible in some way?

**Comment by levkowetz** [comment]
No, this was removed by the design team; from what I understand the reasoning was that it should be possible to refer to any table (and figure) with fragment identifiers, #table-1, #table-2, etc; and in order to do so consistently all tables and figures would need a caption giving at least the table / figure number.

For figures this isn't a big issue, since they can be replaced by bare artwork, but the table layout code is run only for tables, so there isn't a clear workaround.

**Comment by larseggert** [comment]
Too bad. https://tools.ietf.org/html/draft-eggert-irtf-rfc2014bis-05#appendix-A has a table in it that summarizes changes since the last version, and with v3 that now gets a useless caption added to it.

**Comment by cabo** [comment]
Tables are often “bare artwork”.

Grüße, Carsten

**Comment by levkowetz** [comment]
On 2020-11-04 15:32, cabo wrote:
> Tables are often “bare artwork”.

But then you have to format them yourself, rather than use <table> with
sub-elements, no?


	Henrik

**Comment by cabo** [comment]
On 2020-11-04, at 16:07, Henrik Levkowetz <notifications@github.com> wrote:
> 
> 
> 
> 
> On 2020-11-04 15:32, cabo wrote:
> > Tables are often “bare artwork”.
> 
> But then you have to format them yourself, rather than use <table> with
> sub-elements, no?

No, I mean, I’m using the table element (however it is called today) to build my in-paragraph-sequence artwork.  That’s what we did in v2 and what Lars has been asking for.

Grüße, Carsten

**Comment by levkowetz** [comment]
On 2020-11-04 17:20, cabo wrote:
> On 2020-11-04, at 16:07, Henrik Levkowetz <notifications@github.com> wrote:
>> 
>> 
>> 
>> 
>> On 2020-11-04 15:32, cabo wrote:
>> > Tables are often “bare artwork”.
>> 
>> But then you have to format them yourself, rather than use <table> with
>> sub-elements, no?
> 
> No, I mean, I’m using the table element (however it is called today) to build my in-paragraph-sequence artwork.  That’s what we did in v2 and what Lars has been asking for.

Ah.  Right.  And that's a use case that's been dropped in v2.  Ack.


	Henrik

**Comment by cabo** [comment]
On 2020-11-04, at 18:15, Henrik Levkowetz <notifications@github.com> wrote:
> 
> Ah. Right. And that's a use case that's been dropped in v2. Ack.

Given that <texttable> is only deprecated, not removed, maybe that use case is alive and well in v3, just not implemented.

Grüße, Carsten

**Comment by levkowetz** [comment]
Hi Carsten,

On 2020-11-05 11:56, cabo wrote:
> On 2020-11-04, at 18:15, Henrik Levkowetz <notifications@github.com> wrote:
>> 
>> Ah. Right. And that's a use case that's been dropped in v2. Ack.
> 
> Given that <texttable> is only deprecated, not removed, maybe that use case is alive and well in v3, just not implemented.

The v3 formatters only know how to process the non-deprecated elements, so if
you're going to use those, then <texttable> has to be converted to <table>.

As I responded in another thread (I think) yesterday, not taking that path
would have made the deprecation a no-op, and a new round of deprecation
of the same elements would have been needed at some point.


Regards,

	Henrik

**Comment by larseggert** [comment]
So I need to use XML? Markdown can't do this?

Also, related: Can I do this for `~~~` figures?

**Comment by cabo** [comment]
v2 was able to have both tables and artwork without captions.

For v3, Henrik has clarified that this simply isn't implemented for tables; a drop-down to `<texttable>` doesn't seem to work (apparently, as kramdown-rfc current only generates `<texttable>`.
For artwork, leaving out captions continues to work for v3: No anchor, no title.

I consider the lack of support of leaving out captions for tables to be a bug, whether that was an intended bug or not.

**Comment by larseggert** [comment]
But if I need an anchor, I get a title, too?

I'm trying to do a bis for RFC8312, which has a bunch of equations and tables that I want to leave as text. But I also need to refer to them from the text...

**Comment by cabo** [comment]
This:

```

| Lars' Table | with | some | content |
|           1 |    2 |    3 |       4 |
|           5 |    6 |    7 |       8 |


~~~ foo
Ohne alles
~~~


~~~ foo
Mit anchor
~~~
{: #foo}

~~~ foo
Mit title
~~~
{: title="bar"}
```

generates this:

```
                  +=============+======+======+=========+
                  | Lars' Table | with | some | content |
                  +=============+======+======+=========+
                  | 1           | 2    | 3    | 4       |
                  +-------------+------+------+---------+
                  | 5           | 6    | 7    | 8       |
                  +-------------+------+------+---------+

                                  Table 1

   Ohne alles

   Mit anchor

                                  Figure 1

   Mit title

                               Figure 2: bar
```

**Comment by cabo** [comment]
Well, if you need an anchor, I don't see how that can be referenced except by giving the table a number, which needs a caption.

**Comment by larseggert** [comment]
So I guess I need to manually manage anchors and references to them. Argh.

**Comment by cabo** [comment]
As the authors of 8312 apparently did for the equations.
(We need to add equations to RFCXML.)

**Comment by larseggert** [comment]
Yes, we did for 8312 - I was hoping things were better now.

**Comment by cabo** [comment]
So I gather 8312 is not an example for the lost functionality of tables as paragraphs.

It is also an example for a basic equation feature that we could use.

**Comment by larseggert** [comment]
Correct, the table stuff was for another draft.

**Comment by cabo** [comment]
(Of course, I'd write a 10-liner to do this for me and generate markdown to my liking.

I could add a counter feature to kramdown-rfc...)

**Comment by cabo** [comment]
Do you have a pointer to that other draft?

**Comment by larseggert** [comment]
This one wants a table with no caption (in the appendix): https://github.com/larseggert/irtf-rfc2014bis

This one wants captions with something other than "Figure": https://github.com/NTAP/rfc8312bis

**Comment by cabo** [comment]
The latter: Real captions or equation numbers (in line)?

**Comment by larseggert** [comment]
An option for inline would be nice, but I'd settle for being able to change "Figure".

**Comment by cabo** [comment]
"Figure" seems to be pretty much hardcoded in xml2rfc.

**Comment by cabo** [comment]
Oh, and you would want separate counters for figures and equations...

**Comment by jgscudder** [comment]
I take it there's still no way to have tables without captions?

**Comment by cabo** [comment]
On 2022-10-04, at 15:30, jgscudder ***@***.***> wrote:
> 
> 
> I take it there's still no way to have tables without captions?

No, that is a known deficit of RFCXMLv3 that has impacted lots of authors.

RFC 7991:

2.54.  <table>

   Contains a table with a caption with the table number

Please do raise the issue again at ***@***.***; we might be in better luck now with the RFC editor model discussion completed.

Grüße, Carsten
